numba/tests/test_np_functions.py

```diff
@@ -2748,6 +2768,44 @@ def test_repeat_exception(self):
                 with self.assertRaises(TypingError):
                     nbfunc(np.ones(1), rep)
 
+    def test_windowing(self):
+        def check_window(func):
+            np_pyfunc = func
+            np_nbfunc = njit(func)
+
+            for M in [0, 1, 5, 12]:
+                expected = np_pyfunc(M)
+                got = np_nbfunc(M)
+                self.assertPreciseEqual(expected, got)
+
+            with self.assertRaises(TypingError):
+                for M in [-1.1, 0.0, 1.1]:
+                    np_nbfunc(M)
+
+        check_window(np_bartlett)
+        check_window(np_blackman)
+        check_window(np_hamming)
+        check_window(np_hanning)
+
+        # Test np.kaiser separately
+        np_pyfunc = np_kaiser
+        np_nbfunc = njit(np_kaiser)
+
+        for M in [0, 1, 5, 12]:
+            for beta in [0.0, 5.0, 14.0]:
+                expected = np_pyfunc(M, beta)
+                got = np_nbfunc(M, beta)
+
+                if IS_32BITS:
+                    self.assertPreciseEqual(expected, got, prec='double', ulps=2)
+                else:
+                    self.assertPreciseEqual(expected, got, prec='exact')
+
+        with self.assertRaises(TypingError):
```

Comment:
I think that this comment here https://github.com/numba/numba/pull/4076#discussion_r283246433 needs further addressing at this location, whilst this asserts that it's a typing error, it doesn't assert it's the expected typing error. Also, when making this check, recall that the exception handling is in the typing domain, so variations in value have no impact. e.g. a `foo(10.0)` would raise the same error as `foo(20.0)`, as 10.0 and 20.0 would both be seen as floats.

A typical method of doing this might be:
```python
with self.assertRaises(TypingError) as raises:
  np_nbfunc('a', 10)
self.assertIn('M must be an integer', str(raises.exception))
```
then same for `beta`. Thanks.